pkg/unpriv/unpriv.go

```diff
@@ -58,20 +58,16 @@ func splitpath(path string) []string {
 // isNotExist tells you if err is an error that implies that either the path
 // accessed does not exist (or path components don't exist).
 func isNotExist(err error) bool {
```

Comment:
The failure happens because of this change. In particular, `err == unix.ENOTDIR` will always be false. This is what I was referring to:

```go
// isNotExist tells you if err is an error that implies that either the path
// accessed does not exist (or path components don't exist).
func isNotExist(err error) bool {
	if os.IsNotExist(errors.Cause(err)) {
		return true
	}
	// Check that it's not actually an ENOTDIR.
	perr, ok := errors.Cause(err).(*os.PathError)
	if !ok {
		return false
	}
	errno := perr.Err
	return errno == syscall.ENOTDIR || errno == syscall.ENOENT
}
```

(And then just swap `syscall` for `unix`). However, I was thinking about this some more and maybe we should do this (we keep `os.IsNotExist` to avoid possible cases where the stdlib changes in some way).

```go
// isNotExist tells you if err is an error that implies that either the path
// accessed does not exist (or path components don't exist).
func isNotExist(err error) bool {
	if os.IsNotExist(errors.Cause(err)) {
		return true
	}

	// Check that it's not actually an ENOTDIR in disguise. This is technically
	// a dup of doing os.IsNotExist, but it doesn't hurt to be thorough.
	var errno error
	switch err := errors.Cause(err).(type) {
	case *os.PathError:
		errno = err.Err
	case *os.LinkError:
		errno = err.Err
	case *os.SyscallError:
		errno = err.Err
	}
	return errno == syscall.ENOTDIR || errno == syscall.ENOENT
}
```

Also, make sure you don't remove `errors.Cause`s. It's a no-op for non-wrapped error messages, but for wrapper error messages it makes sure that things like `os.IsNotExist` actually do the right thing.